packages/core/src/controllers/ConnectionController.ts

```diff
@@ -252,6 +252,16 @@ export const ConnectionController = {
   async disconnect() {
     const connectionControllerClient = this._getClient()
 
+    const siwx = OptionsController.state.siwx
+    if (siwx) {
+      const activeCaipNetwork = ChainController.getActiveCaipNetwork()
+      const address = ChainController.getActiveCaipAddress()?.split(':')[2] || ''
+
+      if (activeCaipNetwork && address) {
+        siwx.revokeSession(activeCaipNetwork.caipNetworkId, address)
+      }
+    }
```

Comment:
why was there no siwe handling here? 🤔

Reply:
I guess it is because siwe is handled spread around the code, so it is not centralized here